pkg/adapter/nimbus-kyverno/processor/kpbuilder.go

```diff
@@ -239,75 +351,322 @@ func cocoRuntimeAddition(np *v1alpha1.NimbusPolicy) ([]kyvernov1.Policy, error)
 	return kps, multierr.Combine(errs...)
 }
 
-func escapeToHost(np *v1alpha1.NimbusPolicy, rule v1alpha1.Rule) kyvernov1.Policy {
+func virtualPatch(np *v1alpha1.NimbusPolicy, logger logr.Logger) ([]kyvernov1.Policy, error) {
+	rule := np.Spec.NimbusRules[0].Rule
+	requiredCVES := rule.Params["cve_list"]
+	var kps []kyvernov1.Policy
+	resp, err := utils.FetchVirtualPatchData[[]map[string]any]()
+	if err != nil {
+		return kps, err
+	}
+	for _, currObj := range resp {
+		image := currObj["image"].(string)
+		cves := currObj["cves"].([]any)
+		for _, obj := range cves {
+			cveData := obj.(map[string]any)
+			cve := cveData["cve"].(string)
+			if utils.Contains(requiredCVES, cve) {
+				// create generate kyverno policies which will generate the native virtual patch policies based on the CVE's
+				karmorPolCount := 1
+				kyvPolCount := 1
+				netPolCount := 1
+				virtualPatch := cveData["virtual_patch"].([]any)
+				for _, policy := range virtualPatch {
+					pol := policy.(map[string]any)
+					policyData, ok := pol["karmor"].(map[string]any)
+					if ok {
+						karmorPol, err := generatePol("karmor", cve, image, np, policyData, karmorPolCount, logger)
+						if err != nil {
+							logger.V(2).Error(err, "Error while  generating karmor policy")
+						}
+						kps = append(kps, karmorPol)
+						karmorPolCount += 1
+					}
+					policyData, ok = pol["kyverno"].(map[string]any)
+					if ok {
+						kyvernoPol, err := generatePol("kyverno", cve, image, np, policyData, kyvPolCount, logger)
+						if err != nil {
+							logger.V(2).Error(err, "Error while  generating kyverno policy")
+						}
+						kps = append(kps, kyvernoPol)
+						kyvPolCount += 1
+					}
+					
+					policyData, ok = pol["netpol"].(map[string]any)
+					if ok {
+						netPol, err := generatePol("netpol", cve, image, np, policyData, netPolCount, logger)
+						if err != nil {
+							logger.V(2).Error(err, "Error while  generating network policy")
+						}
+						kps = append(kps, netPol)
+						netPolCount += 1
+					}
+				}
+			}
+		}
+	}
+	return kps, nil
+}
 
-	var psa_level api.Level = api.LevelBaseline
-	var matchResourceFilters []kyvernov1.ResourceFilter
+func addManagedByAnnotation(kp *kyvernov1.Policy) {
+	kp.Annotations["app.kubernetes.io/managed-by"] = "nimbus-kyverno"
+}
 
-	if rule.Params["psa_level"] != nil {
+func generatePol(polengine string, cve string, image string, np *v1alpha1.NimbusPolicy, policyData map[string]any, count int, logger logr.Logger) (kyvernov1.Policy, error) {
+	var pol kyvernov1.Policy
+	labels := np.Spec.Selector.MatchLabels
+	cve = strings.ToLower(cve)
+	uid := np.ObjectMeta.GetUID()
+	ownerShipList := []any{
+		map[string]any{
+			"apiVersion":         "intent.security.nimbus.com/v1alpha1",
+			"blockOwnerDeletion": true,
+			"controller":         true,
+			"kind":               "NimbusPolicy",
+			"name":               np.GetName(),
+			"uid":                uid,
+		},
+	}
```

Comment:
Don't you think, we should get the `OwnerReferences` from nimbusPolicy itself instead of hardcoding it?

Ref: https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#ObjectMeta.GetOwnerReferences